src/otx/core/utils/tile_merge.py

```diff
@@ -120,62 +139,153 @@ def merge(
+                feature_vectors.append(tile_entity.feature_vector)
+                saliency_maps.append(tile_entity.saliency_map)
 
         bboxes = torch.stack(bboxes) if len(bboxes) > 0 else torch.empty((0, 4), device=img_info.device)
         labels = torch.stack(labels) if len(labels) > 0 else torch.empty((0,), device=img_info.device)
         scores = torch.stack(scores) if len(scores) > 0 else torch.empty((0,), device=img_info.device)
 
-        bboxes, labels, scores, _ = self.nms_postprocess(
-            bboxes,
-            scores,
-            labels,
-        )
-
-        return DetPredEntity(
-            image=torch.empty(img_size),
-            img_info=img_info,
-            score=scores,
-            bboxes=tv_tensors.BoundingBoxes(
-                bboxes,
-                canvas_size=img_size,
-                format="XYXY",
-            ),
-            labels=labels,
-        )
+        bboxes, labels, scores, _ = self.nms_postprocess(bboxes, scores, labels)
+
+        entity_params = {
+            "image": torch.empty(img_size),
+            "img_info": img_info,
+            "bboxes": tv_tensors.BoundingBoxes(bboxes, canvas_size=img_size, format="XYXY"),
+            "labels": labels,
+            "score": scores,
+        }
+
+        if explain_mode:
+            merged_vector = np.mean(feature_vectors, axis=0)
+            merged_saliency_map = self._merge_saliency_maps(saliency_maps, img_size, tiles_coords)
+            entity_params.update({"feature_vector": merged_vector, "saliency_map": merged_saliency_map})
+
+        return DetPredEntity(**entity_params)
+
+    def _merge_saliency_maps(
+        self,
+        saliency_maps: list[np.array],
+        shape: tuple[int, int],
+        tiles_coords: list[tuple[int, int, int, int]],
+    ) -> np.ndarray:
+        """Merging saliency maps from each tile for PyTorch implementation.
+
+        OV implementation is on ModelAPI side. Unlike ModelAPI implementation,
+        it doesn't have the first tile with resized untiled image.
+
+        Args:
+            saliency_maps: list of saliency maps, shape of each map is (Nc, H, W)
+            shape: shape of the original image
+            tiles_coords: coordinates of tiles
+
+        Returns:
+            Merged saliency map with shape (Nc, H, W)
+        """
+        if len(saliency_maps) == 1:
+            return saliency_maps[0]
+
+        if len(saliency_maps[0].shape) == 1:
+            return np.ndarray([])
+
+        num_classes = saliency_maps[0].shape[0]
+        map_h, map_w = saliency_maps[0].shape[1:]
+
+        image_h, image_w = shape
+        ratio = map_h / min(image_h, self.tile_size), map_w / min(image_w, self.tile_size)
+
+        image_map_h = int(image_h * ratio[0])
+        image_map_w = int(image_w * ratio[1])
+        merged_map = np.zeros((num_classes, image_map_h, image_map_w))
+
+        for i, saliency_map in enumerate(saliency_maps):
```

Comment:
Yes, with a few updates because of the difference in tiling implementation between PyTorch  and OV tasks:
1. In OV in MAPI the first tile is an entire image resized to tile size. While in PyTorch we don't have this first tile with the whole image
2. Different format of tile coordinates: x1,y1,x2,y2 in MAPI and x,y,w,h for PyTorch

I wrote that it's basically the same function in function description.
I could optimize the code but decided to keep the visual similarity of implementations in OTX and MAPI

Reply:
Ok, I'll ask Eugene. It looks like we should adjust the implementation in MAPI

Reply:
Yeah, we talked about it, here is his comment:

`Previously in inference/testing we return the original image as the first tile, this decision is based on current design limitation. That said, this is for PyTorch only. In OpenVINO tiler, it still returns the original image as the first tile. `

Yes, you're right, we should adjust the implementation in MAPI